app/jobs/search_keyword_job.rb

```diff
@@ -5,25 +5,41 @@ class SearchKeywordJob < ApplicationJob
 
   def perform(search_result_id)
     search_result = SearchResult.find(search_result_id)
-    html_code = Search::SearchService.new(
-      keyword: search_result.keyword,
-      search_engine: search_result.search_engine
-    ).search
+    html_code = search(search_result.keyword, search_result.search_engine)
 
-    return mark_as_failed(search_result) unless html_code
+    parse_result = parse(search_result.search_engine, html_code)
 
-    mark_as_completed(search_result, html_code)
-  rescue IcRailsHueyToby::Errors::SearchServiceError
+    set_parse_result(search_result, html_code, parse_result)
+  rescue IcRailsHueyToby::Errors::SearchServiceError, IcRailsHueyToby::Errors::ParseServiceError
     mark_as_failed(search_result)
   end
 
   private
 
-  def mark_as_completed(search_result, html_code)
-    search_result.update html_code: html_code, status: :completed
+  def search(keyword, search_engine)
+    Search::SearchService.new(
+      keyword: keyword,
+      search_engine: search_engine
+    ).search
+  end
+
+  def parse(search_engine, html_code)
+    Parse::ParseService.new(
+      search_engine: search_engine,
+      html_code: html_code
+    ).parse
+  end
+
+  def set_parse_result(search_result, html_code, parse_result)
+    search_result.update(parse_result.merge({ html_code: html_code }))
+    mark_as_completed(search_result)
```

Comment:
This method should not be here 😅 Based on the method name, it is a very unexpected side effect. It would be best to have at line 12:

```ruby
set_parse_result(search_result, html_code, parse_result)
mark_as_completed(search_result)
```

Reply:
Hmmm...Initially, I thought updating the parse result means that the parsing process has been done and we could update the status as completed 💭 Indeed we should separate the updating state to avoid confusion from the method name. Let me update then 👍

Reply:
Fixed [1c1bdbc](https://github.com/nimblehq/ic-rails-huey-toby/pull/37/commits/1c1bdbc3f1dbdf7ee07606b218b9683f5ea9bf39)